File: converted_source/used/uamendmt.pas.java
// unit uamendmt

from conversion_common import *;
import usupport;
import uplant;
import ufiler;
import delphi_compatability;

// const
kStartAmendmentString = "start posing change";
kEndAmendmentString = "end posing change";



class PdPlantDrawingAmendment extends PdStreamableObject {
    public long partID;
    public String typeOfPart;
    public boolean hide;
    public boolean changeColors;
    public boolean propagateColors;
    public TColorRef faceColor;
    public TColorRef backfaceColor;
    public TColorRef lineColor;
    public boolean addRotations;
    public float xRotation;
    public float yRotation;
    public float zRotation;
    public boolean multiplyScale;
    public boolean propagateScale;
    public short scaleMultiplier_pct;
    public short lengthMultiplier_pct;
    public short widthMultiplier_pct;
    public short applyAtAge;
    
    public void create() {
        this.scaleMultiplier_pct = 100;
        this.lengthMultiplier_pct = 100;
        this.widthMultiplier_pct = 100;
        this.xRotation = 0;
        this.yRotation = 0;
        this.zRotation = 0;
        this.faceColor = usupport.support_rgb(50, 200, 50);
        this.backfaceColor = usupport.support_rgb(20, 150, 20);
        this.lineColor = usupport.support_rgb(40, 100, 40);
    }
    
    public void readFromTextFile(TextFile inputFile) {
        String inputLine = "";
        String varName = "";
        String varValue = "";
        
        while (!UNRESOLVED.eof(inputFile)) {
            // plant reads start line; just continue from there
            UNRESOLVED.readln(inputFile, inputLine);
            if ((trim(inputLine) == "")) {
                continue;
            }
            varName = usupport.stringUpTo(inputLine, "=");
            varValue = usupport.stringBeyond(inputLine, "=");
            if (!this.setField(varName, varValue)) {
                break;
            }
        }
        if (UNRESOLVED.pos(uppercase(kEndAmendmentString), uppercase(inputLine)) <= 0) {
            throw new GeneralException.create("Problem: Expected end of posing change.");
        }
    }
    
    public void readFromMemo(TMemo aMemo, long readingMemoLine) {
        String inputLine = "";
        String varName = "";
        String varValue = "";
        
        while (readingMemoLine <= aMemo.Lines.Count - 1) {
            // plant reads start line; just continue from there
            inputLine = aMemo.Lines.Strings[readingMemoLine];
            if ((trim(inputLine) == "")) {
                continue;
            }
            varName = usupport.stringUpTo(inputLine, "=");
            varValue = usupport.stringBeyond(inputLine, "=");
            if (!this.setField(varName, varValue)) {
                break;
            }
            readingMemoLine += 1;
        }
        if (UNRESOLVED.pos(uppercase(kEndAmendmentString), uppercase(inputLine)) <= 0) {
            throw new GeneralException.create("Problem: Expected end of posing change.");
        }
        return readingMemoLine;
    }
    
    public boolean setField(String varName, String varValue) {
        result = false;
        result = true;
        if (UNRESOLVED.pos("part number", varName) > 0) {
            // how deal with problem?
            this.partID = StrToIntDef(varValue, 0);
        } else if (UNRESOLVED.pos("part type", varName) > 0) {
            this.typeOfPart = varValue;
        } else if (UNRESOLVED.pos("hide", varName) > 0) {
            this.hide = usupport.strToBool(varValue);
        } else if (UNRESOLVED.pos("change colors", varName) > 0) {
            // read these for possible backward compatibility later, but don't do anything with them
            // self.changeColors := strToBool(varValue)
            result = true;
        } else if (UNRESOLVED.pos("propagate colors up stem", varName) > 0) {
            // self.propagateColors := strToBool(varValue)
            result = true;
        } else if (UNRESOLVED.pos("front face color", varName) > 0) {
            // self.faceColor := strToIntDef(varValue, 0)
            result = true;
        } else if (UNRESOLVED.pos("back face color", varName) > 0) {
            // self.backfaceColor := strToIntDef(varValue, 0)
            result = true;
        } else if (UNRESOLVED.pos("line color", varName) > 0) {
            // self.lineColor := strToIntDef(varValue, 0)
            result = true;
        } else if (UNRESOLVED.pos("rotate", varName) > 0) {
            this.addRotations = usupport.strToBool(varValue);
        } else if (UNRESOLVED.pos("x rotation", varName) > 0) {
            this.xRotation = StrToInt(varValue) * 1.0;
        } else if (UNRESOLVED.pos("y rotation", varName) > 0) {
            this.yRotation = StrToInt(varValue) * 1.0;
        } else if (UNRESOLVED.pos("z rotation", varName) > 0) {
            this.zRotation = StrToInt(varValue) * 1.0;
        } else if (UNRESOLVED.pos("change scale", varName) > 0) {
            this.multiplyScale = usupport.strToBool(varValue);
        } else if (UNRESOLVED.pos("propagate scale change up stem", varName) > 0) {
            this.propagateScale = usupport.strToBool(varValue);
        } else if (UNRESOLVED.pos("3d object scale multiplier", varName) > 0) {
            this.scaleMultiplier_pct = StrToIntDef(varValue, 100);
        } else if (UNRESOLVED.pos("line length multiplier", varName) > 0) {
            this.lengthMultiplier_pct = StrToIntDef(varValue, 100);
        } else if (UNRESOLVED.pos("line width multiplier", varName) > 0) {
            this.widthMultiplier_pct = StrToIntDef(varValue, 100);
        } else if (UNRESOLVED.pos("apply at age", varName) > 0) {
            // read these for possible backward compatibility later, but don't do anything with them
            // self.applyAtAge := strToIntDef(varValue, 0)
            result = true;
        } else {
            result = false;
        }
        return result;
    }
    
    public void writeToTextFile(TextFile outputFile) {
        writeln(outputFile, kStartAmendmentString);
        writeln(outputFile, "  part number =" + IntToStr(this.partID));
        writeln(outputFile, "  part type =" + this.typeOfPart);
        writeln(outputFile, "  hide =" + usupport.boolToStr(this.hide));
        //
        //  writeln(outputFile, '  change colors =' + boolToStr(changeColors));
        //  writeln(outputFile, '  propagate colors up stem =' + boolToStr(propagateColors));
        //  writeln(outputFile, '  front face color =' + intToStr(faceColor));
        //  writeln(outputFile, '  back face color =' + intToStr(backfaceColor));
        //  writeln(outputFile, '  line color =' + intToStr(lineColor));
        //  
        writeln(outputFile, "  rotate =" + usupport.boolToStr(this.addRotations));
        writeln(outputFile, "  x rotation =" + IntToStr(intround(this.xRotation)));
        writeln(outputFile, "  y rotation =" + IntToStr(intround(this.yRotation)));
        writeln(outputFile, "  z rotation =" + IntToStr(intround(this.zRotation)));
        writeln(outputFile, "  change scale =" + usupport.boolToStr(this.multiplyScale));
        writeln(outputFile, "  propagate scale change up stem =" + usupport.boolToStr(this.propagateScale));
        writeln(outputFile, "  3d object scale multiplier =" + IntToStr(this.scaleMultiplier_pct));
        writeln(outputFile, "  line length multiplier =" + IntToStr(this.lengthMultiplier_pct));
        writeln(outputFile, "  line width multiplier =" + IntToStr(this.widthMultiplier_pct));
        //
        //  writeln(outputFile, '  apply at age =' + intToStr(applyAtAge));
        //  
        writeln(outputFile, kEndAmendmentString);
    }
    
    public void writeToMemo(TMemo aMemo) {
        aMemo.Lines.Add(kStartAmendmentString);
        aMemo.Lines.Add("  part number =" + IntToStr(this.partID));
        aMemo.Lines.Add("  part type =" + this.typeOfPart);
        aMemo.Lines.Add("  hide =" + usupport.boolToStr(this.hide));
        //
        //  aMemo.lines.add('  change colors =' + boolToStr(changeColors));
        //  aMemo.lines.add('  propagate colors up stem =' + boolToStr(propagateColors));
        //  aMemo.lines.add('  front face color =' + intToStr(faceColor));
        //  aMemo.lines.add('  back face color =' + intToStr(backfaceColor));
        //  aMemo.lines.add('  line color =' + intToStr(lineColor));
        //  
        aMemo.Lines.Add("  rotate =" + usupport.boolToStr(this.addRotations));
        aMemo.Lines.Add("  x rotation =" + IntToStr(intround(this.xRotation)));
        aMemo.Lines.Add("  y rotation =" + IntToStr(intround(this.yRotation)));
        aMemo.Lines.Add("  z rotation =" + IntToStr(intround(this.zRotation)));
        aMemo.Lines.Add("  change scale =" + usupport.boolToStr(this.multiplyScale));
        aMemo.Lines.Add("  propagate scale change up stem =" + usupport.boolToStr(this.propagateScale));
        aMemo.Lines.Add("  3d object scale multiplier =" + IntToStr(this.scaleMultiplier_pct));
        aMemo.Lines.Add("  line length multiplier =" + IntToStr(this.lengthMultiplier_pct));
        aMemo.Lines.Add("  line width multiplier =" + IntToStr(this.widthMultiplier_pct));
        //
        //  aMemo.lines.add('  apply at age =' + intToStr(applyAtAge));
        //  
        aMemo.Lines.Add(kEndAmendmentString);
    }
    
    public String getFullName() {
        result = "";
        result = IntToStr(this.partID) + " (" + this.typeOfPart + ")";
        if (this.hide) {
            result = result + ", hidden";
        }
        if (this.changeColors) {
            result = result + ", colored";
        }
        if (this.addRotations) {
            result = result + ", rotated";
        }
        if (this.multiplyScale) {
            result = result + ", scaled";
        }
        if (this.applyAtAge != 0) {
            result = result + " at age " + IntToStr(this.applyAtAge);
        }
        return result;
    }
    
    public void streamDataWithFiler(PdFiler filer, PdClassAndVersionInformationRecord cvir) {
        super.streamDataWithFiler(filer, cvir);
        this.partID = filer.streamLongint(this.partID);
        this.typeOfPart = filer.streamShortString(this.typeOfPart);
        this.hide = filer.streamBoolean(this.hide);
        this.changeColors = filer.streamBoolean(this.changeColors);
        this.propagateColors = filer.streamBoolean(this.propagateColors);
        filer.streamColorRef(this.faceColor);
        filer.streamColorRef(this.backfaceColor);
        filer.streamColorRef(this.lineColor);
        this.addRotations = filer.streamBoolean(this.addRotations);
        this.xRotation = filer.streamSingle(this.xRotation);
        this.yRotation = filer.streamSingle(this.yRotation);
        this.zRotation = filer.streamSingle(this.zRotation);
        this.multiplyScale = filer.streamBoolean(this.multiplyScale);
        this.propagateScale = filer.streamBoolean(this.propagateScale);
        this.scaleMultiplier_pct = filer.streamSmallint(this.scaleMultiplier_pct);
        this.lengthMultiplier_pct = filer.streamSmallint(this.lengthMultiplier_pct);
        this.widthMultiplier_pct = filer.streamSmallint(this.widthMultiplier_pct);
        this.applyAtAge = filer.streamSmallint(this.applyAtAge);
    }
    
    public void classAndVersionInformation(PdClassAndVersionInformationRecord cvir) {
        cvir.classNumber = UNRESOLVED.kPdPlantDrawingAmendment;
        cvir.versionNumber = 0;
        cvir.additionNumber = 0;
    }
    
}

File: converted_source/used/uamendmt.pas.py
# unit uamendmt

from conversion_common import *
import uclasses
import usupport
import uplant
import ufiler
import delphi_compatability

# const
kStartAmendmentString = "start posing change"
kEndAmendmentString = "end posing change"

class PdPlantDrawingAmendment(PdStreamableObject):
    def __init__(self):
        self.partID = 0L
        self.typeOfPart = ""
        self.hide = false
        self.changeColors = false
        self.propagateColors = false
        self.faceColor = TColorRef()
        self.backfaceColor = TColorRef()
        self.lineColor = TColorRef()
        self.addRotations = false
        self.xRotation = 0.0
        self.yRotation = 0.0
        self.zRotation = 0.0
        self.multiplyScale = false
        self.propagateScale = false
        self.scaleMultiplier_pct = 0
        self.lengthMultiplier_pct = 0
        self.widthMultiplier_pct = 0
        self.applyAtAge = 0
    
    def create(self):
        self.scaleMultiplier_pct = 100
        self.lengthMultiplier_pct = 100
        self.widthMultiplier_pct = 100
        self.xRotation = 0
        self.yRotation = 0
        self.zRotation = 0
        self.faceColor = usupport.support_rgb(50, 200, 50)
        self.backfaceColor = usupport.support_rgb(20, 150, 20)
        self.lineColor = usupport.support_rgb(40, 100, 40)
        return self
    
    def readFromTextFile(self, inputFile):
        inputLine = ""
        varName = ""
        varValue = ""
        
        while not UNRESOLVED.eof(inputFile):
            # plant reads start line; just continue from there
            UNRESOLVED.readln(inputFile, inputLine)
            if (trim(inputLine) == ""):
                continue
            varName = usupport.stringUpTo(inputLine, "=")
            varValue = usupport.stringBeyond(inputLine, "=")
            if not self.setField(varName, varValue):
                break
        if UNRESOLVED.pos(uppercase(kEndAmendmentString), uppercase(inputLine)) <= 0:
            raise GeneralException.create("Problem: Expected end of posing change.")
    
    def readFromMemo(self, aMemo, readingMemoLine):
        inputLine = ""
        varName = ""
        varValue = ""
        
        while readingMemoLine <= aMemo.Lines.Count - 1:
            # plant reads start line; just continue from there
            inputLine = aMemo.Lines.Strings[readingMemoLine]
            if (trim(inputLine) == ""):
                continue
            varName = usupport.stringUpTo(inputLine, "=")
            varValue = usupport.stringBeyond(inputLine, "=")
            if not self.setField(varName, varValue):
                break
            readingMemoLine += 1
        if UNRESOLVED.pos(uppercase(kEndAmendmentString), uppercase(inputLine)) <= 0:
            raise GeneralException.create("Problem: Expected end of posing change.")
        return readingMemoLine
    
    def setField(self, varName, varValue):
        result = false
        result = true
        if UNRESOLVED.pos("part number", varName) > 0:
            # how deal with problem?
            self.partID = StrToIntDef(varValue, 0)
        elif UNRESOLVED.pos("part type", varName) > 0:
            self.typeOfPart = varValue
        elif UNRESOLVED.pos("hide", varName) > 0:
            self.hide = usupport.strToBool(varValue)
        elif UNRESOLVED.pos("change colors", varName) > 0:
            # read these for possible backward compatibility later, but don't do anything with them
            # self.changeColors := strToBool(varValue)
            result = true
        elif UNRESOLVED.pos("propagate colors up stem", varName) > 0:
            # self.propagateColors := strToBool(varValue)
            result = true
        elif UNRESOLVED.pos("front face color", varName) > 0:
            # self.faceColor := strToIntDef(varValue, 0)
            result = true
        elif UNRESOLVED.pos("back face color", varName) > 0:
            # self.backfaceColor := strToIntDef(varValue, 0)
            result = true
        elif UNRESOLVED.pos("line color", varName) > 0:
            # self.lineColor := strToIntDef(varValue, 0)
            result = true
        elif UNRESOLVED.pos("rotate", varName) > 0:
            self.addRotations = usupport.strToBool(varValue)
        elif UNRESOLVED.pos("x rotation", varName) > 0:
            self.xRotation = StrToInt(varValue) * 1.0
        elif UNRESOLVED.pos("y rotation", varName) > 0:
            self.yRotation = StrToInt(varValue) * 1.0
        elif UNRESOLVED.pos("z rotation", varName) > 0:
            self.zRotation = StrToInt(varValue) * 1.0
        elif UNRESOLVED.pos("change scale", varName) > 0:
            self.multiplyScale = usupport.strToBool(varValue)
        elif UNRESOLVED.pos("propagate scale change up stem", varName) > 0:
            self.propagateScale = usupport.strToBool(varValue)
        elif UNRESOLVED.pos("3d object scale multiplier", varName) > 0:
            self.scaleMultiplier_pct = StrToIntDef(varValue, 100)
        elif UNRESOLVED.pos("line length multiplier", varName) > 0:
            self.lengthMultiplier_pct = StrToIntDef(varValue, 100)
        elif UNRESOLVED.pos("line width multiplier", varName) > 0:
            self.widthMultiplier_pct = StrToIntDef(varValue, 100)
        elif UNRESOLVED.pos("apply at age", varName) > 0:
            # read these for possible backward compatibility later, but don't do anything with them
            # self.applyAtAge := strToIntDef(varValue, 0)
            result = true
        else:
            result = false
        return result
    
    def writeToTextFile(self, outputFile):
        writeln(outputFile, kStartAmendmentString)
        writeln(outputFile, "  part number =" + IntToStr(self.partID))
        writeln(outputFile, "  part type =" + self.typeOfPart)
        writeln(outputFile, "  hide =" + usupport.boolToStr(self.hide))
        #
        #  writeln(outputFile, '  change colors =' + boolToStr(changeColors));
        #  writeln(outputFile, '  propagate colors up stem =' + boolToStr(propagateColors));
        #  writeln(outputFile, '  front face color =' + intToStr(faceColor));
        #  writeln(outputFile, '  back face color =' + intToStr(backfaceColor));
        #  writeln(outputFile, '  line color =' + intToStr(lineColor));
        #  
        writeln(outputFile, "  rotate =" + usupport.boolToStr(self.addRotations))
        writeln(outputFile, "  x rotation =" + IntToStr(intround(self.xRotation)))
        writeln(outputFile, "  y rotation =" + IntToStr(intround(self.yRotation)))
        writeln(outputFile, "  z rotation =" + IntToStr(intround(self.zRotation)))
        writeln(outputFile, "  change scale =" + usupport.boolToStr(self.multiplyScale))
        writeln(outputFile, "  propagate scale change up stem =" + usupport.boolToStr(self.propagateScale))
        writeln(outputFile, "  3d object scale multiplier =" + IntToStr(self.scaleMultiplier_pct))
        writeln(outputFile, "  line length multiplier =" + IntToStr(self.lengthMultiplier_pct))
        writeln(outputFile, "  line width multiplier =" + IntToStr(self.widthMultiplier_pct))
        #
        #  writeln(outputFile, '  apply at age =' + intToStr(applyAtAge));
        #  
        writeln(outputFile, kEndAmendmentString)
    
    def writeToMemo(self, aMemo):
        aMemo.Lines.Add(kStartAmendmentString)
        aMemo.Lines.Add("  part number =" + IntToStr(self.partID))
        aMemo.Lines.Add("  part type =" + self.typeOfPart)
        aMemo.Lines.Add("  hide =" + usupport.boolToStr(self.hide))
        #
        #  aMemo.lines.add('  change colors =' + boolToStr(changeColors));
        #  aMemo.lines.add('  propagate colors up stem =' + boolToStr(propagateColors));
        #  aMemo.lines.add('  front face color =' + intToStr(faceColor));
        #  aMemo.lines.add('  back face color =' + intToStr(backfaceColor));
        #  aMemo.lines.add('  line color =' + intToStr(lineColor));
        #  
        aMemo.Lines.Add("  rotate =" + usupport.boolToStr(self.addRotations))
        aMemo.Lines.Add("  x rotation =" + IntToStr(intround(self.xRotation)))
        aMemo.Lines.Add("  y rotation =" + IntToStr(intround(self.yRotation)))
        aMemo.Lines.Add("  z rotation =" + IntToStr(intround(self.zRotation)))
        aMemo.Lines.Add("  change scale =" + usupport.boolToStr(self.multiplyScale))
        aMemo.Lines.Add("  propagate scale change up stem =" + usupport.boolToStr(self.propagateScale))
        aMemo.Lines.Add("  3d object scale multiplier =" + IntToStr(self.scaleMultiplier_pct))
        aMemo.Lines.Add("  line length multiplier =" + IntToStr(self.lengthMultiplier_pct))
        aMemo.Lines.Add("  line width multiplier =" + IntToStr(self.widthMultiplier_pct))
        #
        #  aMemo.lines.add('  apply at age =' + intToStr(applyAtAge));
        #  
        aMemo.Lines.Add(kEndAmendmentString)
    
    def getFullName(self):
        result = ""
        result = IntToStr(self.partID) + " (" + self.typeOfPart + ")"
        if self.hide:
            result = result + ", hidden"
        if self.changeColors:
            result = result + ", colored"
        if self.addRotations:
            result = result + ", rotated"
        if self.multiplyScale:
            result = result + ", scaled"
        if self.applyAtAge != 0:
            result = result + " at age " + IntToStr(self.applyAtAge)
        return result
    
    def streamDataWithFiler(self, filer, cvir):
        PdStreamableObject.streamDataWithFiler(self, filer, cvir)
        self.partID = filer.streamLongint(self.partID)
        self.typeOfPart = filer.streamShortString(self.typeOfPart)
        self.hide = filer.streamBoolean(self.hide)
        self.changeColors = filer.streamBoolean(self.changeColors)
        self.propagateColors = filer.streamBoolean(self.propagateColors)
        filer.streamColorRef(self.faceColor)
        filer.streamColorRef(self.backfaceColor)
        filer.streamColorRef(self.lineColor)
        self.addRotations = filer.streamBoolean(self.addRotations)
        self.xRotation = filer.streamSingle(self.xRotation)
        self.yRotation = filer.streamSingle(self.yRotation)
        self.zRotation = filer.streamSingle(self.zRotation)
        self.multiplyScale = filer.streamBoolean(self.multiplyScale)
        self.propagateScale = filer.streamBoolean(self.propagateScale)
        self.scaleMultiplier_pct = filer.streamSmallint(self.scaleMultiplier_pct)
        self.lengthMultiplier_pct = filer.streamSmallint(self.lengthMultiplier_pct)
        self.widthMultiplier_pct = filer.streamSmallint(self.widthMultiplier_pct)
        self.applyAtAge = filer.streamSmallint(self.applyAtAge)
    
    def classAndVersionInformation(self, cvir):
        cvir.classNumber = uclasses.kPdPlantDrawingAmendment
        cvir.versionNumber = 0
        cvir.additionNumber = 0
    
